src/test/kotlin/org/rust/ide/intentions/DemorgansLawIntentionTest.kt

```diff
@@ -37,84 +37,44 @@ class DemorgansLawIntentionTest : RsIntentionTestBase(DemorgansLawIntention::cla
         }
     """)
 
-    fun `test complex 1`() = doAvailableTest("""
+    fun `test complex 1`() = doAvailableSymmetricTest("""
         fn main() {
             if (a && b && c) /*caret*/|| d {}
         }
     """, """
-        fn main() {
-            if !(!(a && b && c) && !d) {}
-        }
-    """)
-
-    fun `test reverse complex 1`() = doAvailableTest("""
         fn main() {
             if !(!(a && b && c) /*caret*/&& !d) {}
         }
-    """, """
-        fn main() {
-            if (a && b && c) || d {}
-        }
     """)
 
-    fun `test complex 2`() = doAvailableTest("""
+    fun `test complex 2`() = doAvailableSymmetricTest("""
         fn main() {
             if (20 >= 50) /*caret*/&& 40 != 20 {}
         }
     """, """
-        fn main() {
-            if !((20 < 50) || 40 == 20) {}
-        }
-    """)
-
-    fun `test reverse complex 2`() = doAvailableTest("""
         fn main() {
             if !((20 < 50) /*caret*/|| 40 == 20) {}
         }
-    """, """
-        fn main() {
-            if (20 >= 50) && 40 != 20 {}
-        }
     """)
 
-    fun `test complex 3`() = doAvailableTest("""
+    fun `test complex 3`() = doAvailableSymmetricTest("""
         fn main() {
             if !(2 + 2 == 2 /*caret*/&& (foo.bar() || !(78 < 90 || 90 > 78)) || (20 >= 50 && ((40 != 20)))) {}
         }
     """, """
         fn main() {
-            if !(!(2 + 2 != 2 || !(foo.bar() || !(78 < 90 || 90 > 78))) || (!(20 < 50 || ((40 == 20))))) {}
+            if !(!(2 + 2 != 2 /*caret*/|| !(foo.bar() || !(78 < 90 || 90 > 78))) || (!(20 < 50 || ((40 == 20))))) {}
         }
     """)
 
-    fun `test reverse complex 3`() = doAvailableTest("""
-        fn main() {
-            if !(!(2 + 2 != 2 /*caret*/|| !(foo.bar() || !(78 < 90 || 90 > 78))) || (!(20 < 50 || !((40 != 20))))) {}
-        }
-    """, """
-        fn main() {
-            if !(2 + 2 == 2 && (foo.bar() || !(78 < 90 || 90 > 78)) || (20 >= 50 && ((40 != 20)))) {}
-        }
-    """)
-
-    fun `test complex 4`() = doAvailableTest("""
+    fun `test complex 4`() = doAvailableSymmetricTest("""
         fn main() {
             if 20 >= 50 /*caret*/&& 40 != 20 {}
         }
     """, """
-        fn main() {
-            if !(20 < 50 || 40 == 20) {}
-        }
-    """)
-
-    fun `test reverse complex 4`() = doAvailableTest("""
         fn main() {
             if !(20 < 50 /*caret*/|| 40 == 20) {}
         }
-    """, """
-        fn main() {
-            if 20 >= 50 && 40 != 20 {}
-        }
     """)
 
     fun `test constant`() = doAvailableTest("""
```

Comment:
Just wondering, what about converting other tests to symmetric ones?

Reply:
Converted